wire/bench_test.go

```diff
@@ -432,6 +434,221 @@ func BenchmarkWriteBlockHeader(b *testing.B) {
 	}
 }
 
+// BenchmarkDecodeGetHeaders performs a benchmark on how long it takes to
+// decode a getheaders message with the maximum number of block locator hashes.
+func BenchmarkDecodeGetHeaders(b *testing.B) {
+	// Create a message with the maximum number of block locators.
+	pver := ProtocolVersion
+	var m MsgGetHeaders
+	for i := 0; i < MaxBlockLocatorsPerMsg; i++ {
+		hash, err := NewShaHashFromStr(fmt.Sprintf("%x", i))
+		if err != nil {
+			b.Fatalf("NewShaHashFromStr: unexpected error: %v", err)
+		}
+		m.AddBlockLocatorHash(hash)
+	}
+
+	// Serialize it so the bytes are available to test the decode below.
+	var bb bytes.Buffer
+	if err := m.BtcEncode(&bb, pver); err != nil {
+		b.Fatalf("MsgGetHeaders.BtcEncode: unexpected error: %v", err)
+	}
+	buf := bb.Bytes()
+
+	b.ResetTimer()
```

Comment:
Shouldn't the timer resets go just before the loop? Otherwise this will include an additional allocs and CPU time for the reader creation and message declaration.